### PR TITLE
vcert playbook - fixes issue with timeout

### PR DESCRIPTION
**Pull request by luispresuelVenafi**
vcert playbook - fixes issue when default time is not set, it was taking 30 secs instead of 180, also added chance to set own timeout

### CODE REVIEW

File: pkg/playbook/app/vcertutil/vcertutil.go [hunk 1]
```diff
@@ -91,6 +96,27 @@ func buildClient(config domain.Config, zone string) (endpoint.Connector, error)
...
+	vcertConfig.Client = &http.Client{
+		Timeout: time.Duration(DefaultTimeout) * time.Second,
+	}
+	if timeout > 0 {
+		vcertConfig.Client.Timeout = time.Duration(timeout) * time.Second
+	}
+	var connectionTrustBundle *x509.CertPool
+
+	if vcertConfig.ConnectionTrust != "" {
+		log.Println("Using trust bundle in custom http client")
```
Comment (rvelaVenafi) [inline pkg/playbook/app/vcertutil/vcertutil.go:108]: use zap.log 
You need to pass it to the function

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

**Review by sbanic-venafi — APPROVED** [review]

File: pkg/playbook/app/vcertutil/vcertutil.go [hunk 2]
```diff
@@ -91,6 +95,28 @@ func buildClient(config domain.Config, zone string) (endpoint.Connector, error)
...
+	vcertConfig.Client = &http.Client{
+		Timeout: time.Duration(DefaultTimeout) * time.Second,
+	}
+	if timeout > 0 {
+		vcertConfig.Client.Timeout = time.Duration(timeout) * time.Second
+	}
+	var connectionTrustBundle *x509.CertPool
+
+	if vcertConfig.ConnectionTrust != "" {
+		zap.L().Debug("Using trust bundle in custom http client")
```
Comment (rvelaVenafi) [inline pkg/playbook/app/vcertutil/vcertutil.go:107]: Don't just invoke zap, pass the actual logger used by the program. This way zap will create a default logger without the customizations from the plyabook

**Review by rvelaVenafi — APPROVED** [review]

File: pkg/playbook/app/vcertutil/vcertutil.go (hunk 2 above)
Comment (rvelaVenafi) [inline pkg/playbook/app/vcertutil/vcertutil.go:107]: Don't just invoke zap, pass the actual logger used by the program. This way zap will create a default logger without the customizations from the plyabook

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

**Review by BeardedPrincess — APPROVED** [review]
This looks good.. are there more changes coming for this? (I only see the one file changed, I'm assuming we'll need to update

File: pkg/playbook/app/vcertutil/vcertutil.go
Comment (BeardedPrincess) [inline pkg/playbook/app/vcertutil/vcertutil.go:1]: @luispresuelVenafi 

Line 76 (in the 'else' statement of if supports synchronous response) is for some reason also setting the timeout on vrequest before taking action. I'm not sure that causes any issue, but it either needs to be updated to use the new timeout variable, or removed if it has no effect. (client is already created, so not sure if it actually does anything)

File: pkg/playbook/app/vcertutil/vcertutil.go [hunk 3]
```diff
@@ -69,7 +73,6 @@ func EnrollCertificate(config domain.Config, request domain.PlaybookRequest) (*c
 		zap.L().Debug("successfully requested certificate", zap.String("requestID", reqID))
 
 		vRequest.PickupID = reqID
-		vRequest.Timeout = 180 * time.Second
```
Comment (rvelaVenafi) [inline pkg/playbook/app/vcertutil/vcertutil.go:72]: If you remove this line, and the first call to retrieve fails (certificate is not in the ISSUED state), you will have to keep invoking the retrieve function or fail. The timeout in the request object is to loop over the certificate retrieval for slow CAs.

Comment (rvelaVenafi) [reply pkg/playbook/app/vcertutil/vcertutil.go:72]: Do not remove it. It makes sense.

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: pkg/playbook/app/vcertutil/vcertutil.go (hunk 3 above)
Comment (luispresuelVenafi) [reply pkg/playbook/app/vcertutil/vcertutil.go:72]: It doesn't make sense, that timeout is already set before

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**